web-ui/src/main/resources/catalog/components/edit/onlinesrc/OnlineSrcDirective.js

```diff
@@ -1944,7 +1950,12 @@
                  * @returns {boolean}
                  */
                 scope.canEnableLinkButton = function (selectRecords) {
-                  if (selectRecords.length < 1) return false;
+                  if (
+                    !selectRecords ||
+                    !Array.isArray(selectRecords) ||
+                    selectRecords.length < 1
+                  )
+                    return false;
 
```

Comment:
Note: This issue fixes the following error that would occur on the edit page.

![image](https://github.com/geonetwork/core-geonetwork/assets/1868233/776cc40d-a36c-4c50-b4d8-ece2784cbc00)